baseband/mark5b/tests/test_mark5b.py

```diff
@@ -285,6 +285,37 @@ def test_header_times(self):
         with pytest.raises(ValueError):
             header.get_time(frame_nr=1)
 
+        # Check setting time using framerate.
+        sample_rate = 128. * u.MHz
+        samples_per_frame = 5000
+        # Max frame_nr is 2**15; this sets it to 25600.
+        frame_rate = sample_rate / samples_per_frame
+        header.set_time(time=(start_time + 1. / frame_rate),
+                        framerate=frame_rate)
+        header.get_time(frame_rate)
+        assert abs(header.get_time(frame_rate) -
+                   start_time - 1. / frame_rate) < 1. * u.ns
+        header.set_time(time=(start_time + 3921. / frame_rate),
+                        framerate=frame_rate)
+        assert abs(header.get_time(frame_rate) -
+                   start_time - 3921. / frame_rate) < 1. * u.ns
+        # Test using bcd_fraction gives us within 0.1 ms accuracy
+        assert abs(header.time - start_time - 3921. / frame_rate) < 0.1 * u.ms
+        header.set_time(time=(start_time + 25599. / frame_rate),
+                        framerate=frame_rate)
+        assert abs(header.get_time(frame_rate) -
+                   start_time - 25599. / frame_rate) < 1. * u.ns
+        # Check rounding when using passing fractional frametimes.
+        header.set_time(time=(start_time + 25598.53 / frame_rate),
+                        framerate=frame_rate)
+        assert abs(header.get_time(frame_rate) -
+                   start_time - 25599. / frame_rate) < 1. * u.ns
+        # Check rounding to the nearest second when less than 2 ns away.
+        header.set_time(time=(start_time + 1.8 * u.ns), frame_nr=0)
```

Comment:
So, really this could now be quite a bit worse, correct? Essentially half a frame?

Reply:
Yes as with VDIF's `set_time`, though weirdly with the new code that test should fail!  Having a look now.